ribs/schedulers/_bandit_scheduler.py

```diff
@@ -258,17 +264,24 @@ def ask(self):
         #   terminated/restarted will be reselected. Otherwise, if reselect is
         #   "all", then all emitters are reselected.
         if reselect.any():
-            ucb1 = np.full_like(self._emitter_pool, np.inf)
+            ucb1 = np.full_like(
+                self._emitter_pool, np.inf
+            )  # np.inf forces to select emitters that were not yet selected
             update_ucb = self._selection != 0
             if update_ucb.any():
                 ucb1[update_ucb] = (
                     self._success[update_ucb] / self._selection[update_ucb] +
                     self._zeta * np.sqrt(
                         np.log(self._success.sum()) /
                         self._selection[update_ucb]))
-            # Activate top emitters based on UCB1.
-            activate = np.argsort(ucb1)[-reselect.sum():]
-            self._active_arr[activate] = True
+            # Activate top emitters based on UCB1, until there are num_active
+            # active emitters. Activate only inactive emitters.
+            activate = np.argsort(ucb1)[::-1]
+            for i in activate:
+                if self._active_arr.sum() == self._num_active:
```

Comment:
Would you mind adding a variable that counts the number of active emitters so that we don't keep calling `.sum()` here?